src/test/java/roomescape/e2e/ReservationApiTest.java

```diff
@@ -0,0 +1,136 @@
+package roomescape.e2e;
+
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.test.annotation.DirtiesContext;
+import roomescape.dto.ReservationResponse;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.CoreMatchers.is;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+public class ReservationApiTest {
+
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    @BeforeEach
+    void setDB() {
+        jdbcTemplate.execute("DELETE FROM reservation");
+        jdbcTemplate.execute("DELETE FROM reservation_time");
+        jdbcTemplate.execute("ALTER TABLE reservation ALTER COLUMN id RESTART WITH 1");
+        jdbcTemplate.execute("ALTER TABLE reservation_time ALTER COLUMN id RESTART WITH 1");
+        jdbcTemplate.update("INSERT INTO reservation_time (start_at) VALUES (?)", "15:40");
+        jdbcTemplate.update("INSERT INTO reservation (name, date, time_id) VALUES (?, ?, ?)", "브라운", "2023-08-05", 1L);
+    }
+
+    @Test
+    @DisplayName("예약 추가 시 예약 정보를 저장하고 200 OK를, 삭제 시 id에 해당하는 예약을 제거하고 200 OK를 응답해야 한다")
+    void level3() {
+        jdbcTemplate.update("INSERT INTO reservation_time(start_at) values (?)", "11:00");
+        Map<String, String> params = new HashMap<>();
+        params.put("name", "브라운");
+        params.put("date", "2023-08-05");
+        params.put("timeId", "2");
```

Comment:
```suggestion
        ReservationRequest reservationRequest = new ReservationRequest("브라운", LocalDate.of(2023, 8, 5), 2L);
```

Dto를 사용해서 body를 만들어주는 것도 가능합니다.

Reply:
dto를 사용할 수 있었군요...! Map을 사용하는 것 보다 더 편리해서 좋은 것 같습니다.
Map을 사용하던 로직에 모두 적용 완료했습니다!

```java
...
ReservationRequest request = new ReservationRequest("브라운", LocalDate.parse("2023-08-05"), 2L);

RestAssured.given().log().all()
...
```